hoomd/architecture.rst

```diff
@@ -0,0 +1,67 @@
+ARCHITECTURE.rst
```

Comment:
Why are we adding this in `ARCHITECTURE.rst`? Is there something wrong with the existing `ARCHITECTURE.md` file? I would prefer not to maintain two separate files with similar goals.

Reply:
Honestly I somehow just missed it, I will fix that immediately.